build.rs

```diff
@@ -66,95 +66,261 @@
-            let m = if target_triple[2].contains("bsd") {
-                "gmake"
-            } else {
-                "make"
-            };
-            std::ffi::OsString::from(m)
-        });
-        run_command_with_args(&make, &args);
-    } else {
-        build_msvc(&target_triple, disable_opt, &num_jobs, lib_path.clone(),
-                   out_dir);
-    }
+const RING_INLINE_FILES: &'static [&'static str] =
+    &["crypto/ec/ecp_nistz256_table.inl",
+      "crypto/ec/ecp_nistz384.inl",
+      "crypto/ec/gfp_limbs.inl",
+      "crypto/ec/ecp_nistz384_mul.inl"];
 
-    println!("cargo:rustc-link-search=native={}", lib_path.to_str().unwrap());
-    println!("cargo:rustc-link-lib=static={}-core", LIB_NAME);
+#[cfg_attr(rustfmt, rustfmt_skip)]
+const RING_PERL_INCLUDES: &'static [&'static str] =
+    &["crypto/sha/asm/sha-x86_64.pl",
+      "crypto/sha/asm/sha-armv8.pl",
+      "crypto/perlasm/x86masm.pl",
+      "crypto/perlasm/x86gas.pl",
+      "crypto/perlasm/x86nasm.pl",
+      "crypto/perlasm/x86asm.pl",
+      "crypto/perlasm/x86_64-xlate.pl",
+      "crypto/perlasm/arm-xlate.pl"];
 
-    // XXX: Ideally, this would only happen for `cargo test`, but we don't know
-    // how to do that yet.
-    println!("cargo:rustc-link-lib=static={}-test", LIB_NAME);
-    if !use_msbuild {
-        println!("cargo:rustc-flags=-l dylib=stdc++");
+#[cfg_attr(rustfmt, rustfmt_skip)]
+const RING_MAKEFILES: &'static [&'static str] =
+    &["Makefile",
+      "mk/top_of_makefile.mk",
+      "mk/bottom_of_makefile.mk",
+      "mk/ring.mk"];
+
+const RING_BUILD_FILE: &'static [&'static str] = &["build.rs"];
+
+#[cfg_attr(rustfmt, rustfmt_skip)]
+const C_FLAGS: &'static [&'static str] =
+    &["-std=c1x",
+      "-Wbad-function-cast",
+      "-Wmissing-field-initializers",
+      "-Wmissing-prototypes",
+      "-Wnested-externs",
+      "-Wstrict-prototypes"];
+
+const CXX_FLAGS: &'static [&'static str] = &["-std=c++0x"];
+
+#[cfg_attr(rustfmt, rustfmt_skip)]
+const CPP_FLAGS: &'static [&'static str] =
+    &["-fpic",
+      "-fdata-sections",
+      "-ffunction-sections",
+      "-pedantic",
+      "-pedantic-errors",
+      "-Wall",
+      "-Werror",
+      "-Wextra",
+      "-Wcast-align",
+      "-Wcast-qual",
+      "-Wenum-compare",
+      "-Wfloat-equal",
+      "-Wformat=2",
+      "-Winline",
+      "-Winvalid-pch",
+      "-Wmissing-declarations",
+      "-Wmissing-field-initializers",
+      "-Wmissing-include-dirs",
+      "-Wredundant-decls",
+      "-Wshadow",
+      "-Wsign-compare",
+      "-Wundef",
+      "-Wuninitialized",
+      "-Wwrite-strings",
+      "-fno-strict-aliasing",
+      "-fvisibility=hidden",
+      "-Wno-cast-align"];
+
+const LD_FLAGS: &'static [&'static str] = &[];
+
+const AS_FLAGS: &'static [&'static str] = &["-Wa,--noexecstack,-gdwarf"];
+
+
+fn main() {
+    for (key, value) in env::vars() {
+        println!("{}: {}", key, value);
     }
 
-    Ok(())
+    check_all_files_tracked();
```

Comment:
The parallelism commit should also make `check_all_files_tracked()` function work in parallel.

Also, I suggest that we put the `check_all_files_tracked()` call after `build_c_code();`. Imagine that there is an error in the C code; we'd rather get that error sooner, rather than wait for the `check_all_files_tracked()` function.

And/or, is there a way to have Rayon do the `check_all_files_tracked()` work in parallel with `build_c_code()`, but giving tasks from `build_c_code()` higher priority?